pkg/rtc/participant.go

```diff
@@ -344,22 +344,26 @@ func (p *ParticipantImpl) SetMetadata(metadata string) {
 }
 
 func (p *ParticipantImpl) ClaimGrants() *auth.ClaimGrants {
-	return p.grants.Load().(*auth.ClaimGrants)
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	return p.grants
```

Comment:
This is not great. This is returning a pointer and the underlying data could change while that pointer fields are being used. A deep copied object return would be good. Not sure how to do deep copy (we will need equivalent of `proto.Clone`). What is the standard way to do that?

Reply:
I will look at implementing a clone method. Not very concerned about somebody using this API modifying it (although that is a concern). It is currently used only in room manager to refresh token.

I am more concerned about some call site holding that pointer and using it at inopportune time. Biggest concern is that underlying structure is being modified and it has some inconsistent values and it gets used in calculating token or something like that. Will make a Clone method in a separate PR.